iceoryx_binding_c/include/iceoryx_binding_c/subscriber.h

```diff
@@ -80,4 +81,23 @@ bool iox_sub_has_new_chunks(iox_sub_t const self);
 /// @return true if there are lost chunks otherwise false
 bool iox_sub_has_lost_chunks(iox_sub_t const self);
 
+/// @brief attaches the subscriber to a waitset
+/// @param[in] self handle to the subscriber
+/// @param[in] waitset handle to the waitset
+/// @param[in] event the type of the event which should be attached to the waitset
+/// @param[in] triggerId the user defined trigger id
+/// @param[in] callback a callback which is attached to the trigger
+/// @return if it is attached was successfully it return WaitSetResult_SUCCESS
+///         otherwise an enum which is describing the error
+ENUM iox_WaitSetResult iox_sub_attach_to_ws(iox_sub_t const self,
+                                            iox_ws_t const waitset,
+                                            const ENUM iox_SubscriberEvent event,
+                                            const uint64_t triggerId,
+                                            void (*callback)(iox_sub_t));
+
+/// @brief detaches the subscriber from a waitset
+/// @param[in] self handle to the subscriber
+void iox_sub_detach_ws(iox_sub_t const self);
```

Comment:
Yes of course. I also thought of this during the weekend and I would add the parameter right away so that later the public API does not change anymore.

Reply:
Then you should do it now to not break the API later

Reply:
Renamed it to `iox_sub_detach_of(iox_sub_t const self, iox_SubscriberEvent event)`